### ISSUE TITLE
0.5.2 iOS的问题

### ISSUE THREAD

**Issue by pleasecallmelm** [creativecreatorormaybenot/wakelock#122]
你这个也太硬核了。那个观察者是全局的，会导致别的插件，例如播放器内部设置idleTimerDisabled为true，调用这个idleTimerDisabled 的set函数时触发观察者，从而设置回false。建议改成下面这样。`- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context {
  if (context == mKeyPathObserverContextApplicationIsIdleTimerDisabled) {
    if (UIApplication.sharedApplication.idleTimerDisabled != self.enable) {
    //  UIApplication.sharedApplication.idleTimerDisabled = self.enable;
        self.enable = UIApplication.sharedApplication.idleTimerDisabled;
    }
  } else {
    [super observeValueForKeyPath:keyPath ofObject:object change:change context:context];
  }
}`

**Comment by creativecreatorormaybenot** [comment]
Hi @pleasecallmelm, unfortunately my Chinese skills have degraded in the past years, so I cannot tell what exactly the issue is about. Thanks for opening it - can you rephrase it in English? 🙂
